### PR TITLE
Fix import AIP compression algorithm flag

### PR DESCRIPTION
**Pull request by Dhwaniartefact**
Connected to https://github.com/archivematica/Issues/issues/1516

### CODE REVIEW

File: storage_service/common/tests/test_command_import_aip.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+
+# Test if package is compressed or uncompressed
+@pytest.mark.parametrize(
+    "compression_algorithm, expected_compression_algorithm",
+    [
+        # matching for compressed package
+        ("7z with bzip", True),
+        # matching for uncompressed package
+        ("7z without compression", True),
```
Comment (replaceafill) [inline storage_service/common/tests/test_command_import_aip.py]: Why is it `expected_compresion_algorithm` a parameter if it's always `True`?

File: storage_service/common/tests/test_command_import_aip.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    call_command(
+        "import_aip",
+        "--decompress-source",
+        "--aip-storage-location",
+        aip_location.uuid,
+        AIP_PATH,
+    )
+    captured = capsys.readouterr()
+    # shutil.rmtree(AIP_PATH.split("import_aip_bag")[-1])
+    assert "Successfully imported AIP" in captured.out
```
Comment (replaceafill) [inline storage_service/common/tests/test_command_import_aip.py]: Should this test also assert that the package was created and that is not compressed like the other tests do?

File: storage_service/common/tests/test_command_import_aip.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    mocker.patch("pwd.getpwnam")
+    call_command(
+        "import_aip",
+        "--decompress-source",
+        "--aip-storage-location",
+        aip_location.uuid,
+        AIP_PATH,
+    )
+    captured = capsys.readouterr()
+    # shutil.rmtree(AIP_PATH.split("import_aip_bag")[-1])
```
Comment (replaceafill) [inline storage_service/common/tests/test_command_import_aip.py]: Is this comment necessary?

File: storage_service/common/tests/test_command_import_aip.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+from django.core.management import call_command
+from locations import models
+
+TEST_DIR = os.path.dirname(os.path.realpath(__file__))
+FIXTURES_DIR = os.path.join(TEST_DIR, "fixtures")
+AIP_PATH = os.path.join(FIXTURES_DIR, "import_aip_test.7z")
+
+
+@pytest.fixture
+def aip_location(db, tmp_path):
```
Comment (replaceafill) [inline storage_service/common/tests/test_command_import_aip.py]: Would `aip_storage_location` be a better name for this fixture?

File: storage_service/common/tests/test_command_import_aip.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    space_temp = tmp_path / "sub"
```
Comment (replaceafill) [inline storage_service/common/tests/test_command_import_aip.py]: What is the meaning of `_temp` here?

File: storage_service/common/tests/test_command_import_aip.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    models.Location.objects.create(
+        space=space, purpose=models.Location.STORAGE_SERVICE_INTERNAL, relative_path=""
+    )
+    models.LocalFilesystem.objects.create(space=space)
+    models.LocationPipeline.objects.get_or_create(pipeline=pipeline, location=aipstore)
+    return aipstore
+
+
+@pytest.mark.django_db
+def test_import_aip(mocker, capsys, aip_location):
```
Comment (replaceafill) [inline storage_service/common/tests/test_command_import_aip.py]: I notice the other tests have names such as `test_import_aip_compressed` and `test_import_aip_uncompressed`. Could this test be named in a similar way stating the aspect that is testing?

File: storage_service/common/tests/test_command_import_aip.py [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+        "--aip-storage-location",
+        aip_location.uuid,
+        AIP_PATH,
+    )
+    captured = capsys.readouterr()
+    # shutil.rmtree(AIP_PATH.split("import_aip_bag")[-1])
+    assert "Successfully imported AIP" in captured.out
+
+
+# Test if package is compressed
```
Comment (replaceafill) [inline storage_service/common/tests/test_command_import_aip.py]: I find this comment and the name of test misleading from what it actually tests. Are you testing that the command received a compressed package? You could also expand the name of the test function and not need this comment anymore.

File: storage_service/common/tests/test_command_import_aip.py [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+    # Verify package is not compressed.
+    package = models.Package.objects.first()
+    assert package.is_compressed
+
+
+# Test if package is compressed or uncompressed
+@pytest.mark.parametrize(
+    "compression_algorithm, expected_compression_algorithm",
+    [
+        # matching for compressed package
```
Comment (replaceafill) [inline storage_service/common/tests/test_command_import_aip.py]: Could these comments be full sentences? It's a bit hard to understand them.

File: storage_service/common/tests/test_command_import_aip.py [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+
+    # Verify a new package was created.
+    assert models.Package.objects.count() == 1
+
+    # Verify package is not compressed.
+    package = models.Package.objects.first()
+    assert package.is_compressed
+
+
+# Test if package is compressed or uncompressed
```
Comment (replaceafill) [inline storage_service/common/tests/test_command_import_aip.py]: Same as above, I find this comment and the name of test misleading from what it actually tests. Are you testing that the command received a compressed package again?

Also from this comment it seems we're re-testing what the previous test did?

**Review by replaceafill — COMMENTED** [review]
Looks good overall @Dhwaniartefact. I left a few questions and suggestions on the new test.

File: storage_service/common/tests/test_command_import_aip.py (hunk 3 above)
Comment (Dhwaniartefact) [reply storage_service/common/tests/test_command_import_aip.py]: Thanks for reminding. Forgot to delete it.

File: storage_service/common/tests/test_command_import_aip.py (hunk 5 above)
Comment (Dhwaniartefact) [reply storage_service/common/tests/test_command_import_aip.py]: Creating temporary directory of space.

File: storage_service/common/tests/test_command_import_aip.py (hunk 6 above)
Comment (Dhwaniartefact) [reply storage_service/common/tests/test_command_import_aip.py]: It tests whether AIP is importing. Can you please suggest any other names?

File: storage_service/common/tests/test_command_import_aip.py (hunk 4 above)
Comment (Dhwaniartefact) [reply storage_service/common/tests/test_command_import_aip.py]: sure.

File: storage_service/common/tests/test_command_import_aip.py (hunk 7 above)
Comment (Dhwaniartefact) [reply storage_service/common/tests/test_command_import_aip.py]: sounds good.

File: storage_service/common/tests/test_command_import_aip.py (hunk 1 above)
Comment (Dhwaniartefact) [reply storage_service/common/tests/test_command_import_aip.py]: got u

File: storage_service/common/tests/test_command_import_aip.py (hunk 3 above)
Comment (Dhwaniartefact) [reply storage_service/common/tests/test_command_import_aip.py]: removed

File: storage_service/common/tests/test_command_import_aip.py (hunk 2 above)
Comment (Dhwaniartefact) [reply storage_service/common/tests/test_command_import_aip.py]: sure let me add that assertion.

File: storage_service/common/tests/test_command_import_aip.py (hunk 8 above)
Comment (Dhwaniartefact) [reply storage_service/common/tests/test_command_import_aip.py:83]: # matching compression algorithm flag if package is compressed

Will this work?

Comment (Dhwaniartefact) [reply storage_service/common/tests/test_command_import_aip.py]: sure let me change it

File: storage_service/common/tests/test_command_import_aip.py (hunk 9 above)
Comment (Dhwaniartefact) [reply storage_service/common/tests/test_command_import_aip.py]: ok let me check it

File: storage_service/common/tests/test_command_import_aip.py [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+    models.Location.objects.create(
+        space=space, purpose=models.Location.STORAGE_SERVICE_INTERNAL, relative_path=""
+    )
+    models.LocalFilesystem.objects.create(space=space)
+    models.LocationPipeline.objects.get_or_create(pipeline=pipeline, location=aipstore)
+    return aipstore
+
+
+@pytest.mark.django_db
+def test_import_aip_output(mocker, capsys, aip_storage_location):
```
Comment (replaceafill) [inline storage_service/common/tests/test_command_import_aip.py]: If the test below tests that the package has been compressed, is it worth to test here that it isn't? And if so, should the name of the test be updated accordingly?

File: storage_service/common/tests/test_command_import_aip.py [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+    assert models.Package.objects.count() == 1
+
+
+# Test if package is compressed or uncompressed
+@pytest.mark.parametrize(
+    "compression_algorithm",
+    [
+        # Matching compression algorithm flag if package is compressed
+        ("7z with bzip"),
+        # Matching compression algorithm flag if package is uncompressed
```
Comment (replaceafill) [inline storage_service/common/tests/test_command_import_aip.py]: The last line of this tests asserts that the package is compressed, is this comment outdated now?

File: storage_service/common/tests/test_command_import_aip.py [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+        "--compression-algorithm",
+        compression_algorithm,
+    )
+    captured = capsys.readouterr()
+    assert "Successfully imported AIP" in captured.out
+
+    # Verify a new package was created.
+    assert models.Package.objects.count() == 1
+
+    # Verify package is not compressed.
```
Comment (replaceafill) [inline storage_service/common/tests/test_command_import_aip.py]: The last line of this tests asserts that the package is compressed, is this comment outdated now?

File: storage_service/common/tests/test_command_import_aip.py (hunk 9 above)
Comment (replaceafill) [reply storage_service/common/tests/test_command_import_aip.py]: Is the `uncompressed` part of this comment still relevant?

**Review by replaceafill — APPROVED** [review]
This seems fine now @Dhwaniartefact. I think you only need to update some comments in the test or consider removing them if they don't add any value.

Also, please don't squash your commits once someone has started reviewing your pull request unless they ask you. It makes following reviews difficult. You can squash at the end once the PR is approved and ready to be merged.

File: storage_service/common/tests/test_command_import_aip.py (hunk 10 above)
Comment (Dhwaniartefact) [reply storage_service/common/tests/test_command_import_aip.py]: sure. Thanks

File: storage_service/common/tests/test_command_import_aip.py (hunk 11 above)
Comment (Dhwaniartefact) [reply storage_service/common/tests/test_command_import_aip.py]: got your point. Let me change it.

File: storage_service/common/tests/test_command_import_aip.py (hunk 12 above)
Comment (Dhwaniartefact) [reply storage_service/common/tests/test_command_import_aip.py]: oh yes

**Pull request merged**